### PR TITLE
Parser adjustments v1.1

### PR DESCRIPTION
**Pull request by Filari02**
created async parsing from eshops. now it is available to parse one specific shop or parse item category (for example, if we choose to parse ItemCategory.Laptop, then every shop which is selling laptops will be parsed and return value will be a List of laptops).

implemented requirements:
- asynchronous programming (using async, await)
- Generic methods
- DI

### CODE REVIEW

**Review by Ugniusst — APPROVED** [review]

File: ShopParser/ComputerParsers/AvitelaComputerParser.cs [hunk 1]
```diff
@@ -70,14 +70,15 @@ public List<Computer> ParseShop()
 
 
         //parses laptop window, updates computer fields
-        public Computer ParseWindow(string url)
+        public async Task<Computer> ParseWindow(string url)
```
Comment (Tomas-Per) [inline ShopParser/ComputerParsers/AvitelaComputerParser.cs:73]: No await operator, so the async is kind of redundant

**Review by Tomas-Per — COMMENTED** [review]
Functions have async keywords but do not have await operators, doesn't this just make the async redundant?

File: ShopParser/ComputerParsers/AvitelaComputerParser.cs (hunk 1 above)
Comment (Filari02) [reply ShopParser/ComputerParsers/AvitelaComputerParser.cs:73]: it is redundant for now. but i am planning to optimize parsing even more, so several one shop Windows will be parsed at the same time. there i will use await keyword. if my optimization will not work, i will delete async from here

**Comment by Filari02** [comment]
it is redundant for now. but i am planning to optimize parsing even more, so several one shop Windows will be parsed at the same time. there i will use await keyword. if my optimization will not work, i will delete async from these methods

**Comment by Tomas-Per** [comment]
I guess for now we will leave it as is then.

**Review by Tomas-Per — APPROVED** [review]

**Pull request merged**